### ISSUE TITLE
User configurable content filter and removal of response length limit

### ISSUE THREAD

**Issue by thephoeron** [microsoft/vscode-copilot-release#215]
Running up against the "Oops, your response got filtered..." error message quite a lot today, on the latest Insiders build and extension versions, and have reviewed open issues from over the course of the last two weeks reporting variations of the same. My specific issue appears to be _response filtered because of response length_, which has already been reported several times as a bug. So I've put some thought into feature requests instead which would address these issues.

What I would like is more transparency into and control over the content filter, by enabling inspection of the default content filter rules under Copilot settings on GitHub, the ability to configure and selectively disable groups of them for my account as a whole, and ideally, override those settings at the system, user, and project level within VSCode.

It doesn't really make sense to lump response length limits, which are a technical limitation, with content filters, which are intended to be for some degree of user safety from the system generating known dangerous code or offensive content. It would make more sense to fully differentiate between response length limits and content filters, in settings as well as the interface. The current error message isn't meaningful because it's too overloaded.

Ultimately, as Copilot is a paid service and as part of the private beta for Copilot Chat we're meant to be stress-testing it, I would prefer no limit to response length at all within the token limits of GPT-4, and in such extreme cases, some sort of predictive analysis of a prompt included as part of its context, where Copilot Chat responds as unable to answer followed by a meaningful explanation of why the response length is expected to be too long, rather than answering in the affirmative only to be cut off.

A middle-of-the-road solution is also possible, if Copilot Chat supported a sequence of responses in messages for a single prompt, to give users the experience of no arbitrary limit to response length while keeping each logical block of the response to a manageable size for the sake of token limits.

In summary, I would like:

- predictive analysis of prompts for including expected response size as part of context
- error messages replaced by Copilot Chat clearly explaining why it is unable to answer
- response length limit distinguished from content filter rules
- inspection and configuration of content filter rules, at account, system, user, and project levels
- ultimate removal of response length limit, which could be implemented on top of predictive prompt analysis to generate a sequence of responses per prompt

**Comment by bhavyaus** [comment]
Thanks for creating this issue! We figured it's covering the same as another one we already have: https://github.com/microsoft/vscode-copilot-release/issues/2625. Please refer to that one for updates and discussions. Feel free to open a new issue if you think this is a different problem.
